recipes/longhorn.yaml

```diff
@@ -0,0 +1,44 @@
+apiVersion: v1
+kind: kbrew
+app:
+  repository:
+    name: longhorn
+    url: https://charts.longhorn.io
+    type: helm
+  namespace: "longhorn-system"
+  pre_install:
+  - steps:
+    - curl -s https://raw.githubusercontent.com/longhorn/longhorn/v1.1.1/scripts/environment_check.sh | bash
+    - kubectl apply -f https://raw.githubusercontent.com/longhorn/longhorn-manager/master/deploy/prerequisite/longhorn-iscsi-installation.yaml
+  post_install:
+  - steps:
+    - echo ${USER}:$(echo ${PASSWORD} | openssl passwd -stdin -apr1) >> auth
+    - kubectl -n longhorn-system create secret generic basic-auth --from-file=auth
+    - |
+      echo "
+      apiVersion: networking.k8s.io/v1
+      kind: Ingress
+      metadata:
+        name: longhorn-ingress
+        namespace: longhorn-system
+        annotations:
+          # type of authentication
+          nginx.ingress.kubernetes.io/auth-type: basic
+          # prevent the controller from redirecting (308) to HTTPS
+          nginx.ingress.kubernetes.io/ssl-redirect: 'false'
+          # name of the secret that contains the user/password definitions
+          nginx.ingress.kubernetes.io/auth-secret: basic-auth
+          # message to display with an appropriate context why the authentication is required
+          nginx.ingress.kubernetes.io/auth-realm: 'Authentication Required '
+      spec:
+        rules:
+        - http:
+            paths:
+            - path: /
+              pathType: Prefix
+              backend:
+                service:
+                  name: longhorn-frontend
+                  port:
+                    number: 80
+      " | kubectl -n longhorn-system create -f -
```

Comment:
@sahil-lakhwani can we add a post_install step to validate if the final installation is working as expected

Reply:
add a post_install steps to verify volume provisioning and a simple pod to use that volume. Also added pre_cleanup steps to remove that volume and pod